core/src/main/scala/kafka/consumer/ZookeeperConsumerConnector.scala

```diff
@@ -507,6 +507,10 @@ private[kafka] class ZookeeperConsumerConnector(val config: ConsumerConfig,
       // The child change watchers will be set inside rebalance when we read the children list.
     }
 
+    @throws(classOf[RuntimeException])
+    override def handleSessionEstablishmentError(error: Throwable): Unit = {
+      throw new RuntimeException("zkClient reconnection failed, is zookeeper having issues?", error)
```

Comment:
You shouldn't throw RuntimeException from here. If you do, you're just swallowing up important exceptions. 

IMHO you should explicitly call exit(-1) here, or Utils.croak() function and allow KafkaServer to do the clean start.

Please look the JIRA ticket for more info.